### PR TITLE
AT-5578 xss attack validate name for dev new user

### PR DESCRIPTION
**Pull request by jGonzalez-CCPO**
https://ccpo.atlassian.net/browse/AT-5578

### CODE REVIEW

File: atat/routes/dev.py [hunk 1]
```diff
@@ -183,23 +188,36 @@ def get_or_create_dev_persona(persona):
...
+    first_name = request.args.get("first_name", None, type=str)
+    last_name = request.args.get("last_name", None, type=str)
+    dod_id = request.args.get("dod_id", None, type=str)
 
+    # 1: It need to check fist that have all the parameters
     if None in [first_name, last_name, dod_id]:
         raise IncompleteInfoError()
 
+    # This interphase convert dictionaries into Objects instance. It is Require for step 2.
+    class Struct(object):
```
Comment (kylelaker-ccpo) [inline atat/routes/dev.py:200]: Was there a specific reason to create a `Struct` class over using the standard library--provided `types.SimpleNamespace`? It works almost exactly the same way as what has been done here.

File: atat/routes/dev.py [hunk 2]
```diff
@@ -183,23 +188,36 @@ def get_or_create_dev_persona(persona):
...
+        def __init__(self, **entries):
+            self.__dict__.update(entries)
+
+    # 2: Using standard validator to validate inputs or throw a error
+    is_name({}, Struct(**{"data": first_name}))
+    is_name({}, Struct(**{"data": last_name}))
+    is_number({}, Struct(**{"data": dod_id}))
```
Comment (kylelaker-ccpo) [inline atat/routes/dev.py:207]: The `entries` parameter for `Struct.__init__` already has `**` so doing this on both ends makes the code harder to understand. The symbols are confusing so maybe consider:

```suggestion
    is_name({}, Struct(data=first_name))
    is_name({}, Struct(data=last_name))
    is_number({}, Struct(data=dod_id))
```

File: atat/routes/dev.py [hunk 3]
```diff
@@ -183,23 +188,36 @@ def get_or_create_dev_persona(persona):
 
 @local_access_bp.route("/dev-new-user")
 def dev_new_user():
-    first_name = request.args.get("first_name", None)
-    last_name = request.args.get("last_name", None)
-    dod_id = request.args.get("dod_id", None)
+    first_name = request.args.get("first_name", None, type=str)
+    last_name = request.args.get("last_name", None, type=str)
+    dod_id = request.args.get("dod_id", None, type=str)
```
Comment (kylelaker-ccpo) [inline atat/routes/dev.py:193]: Unless I am mistaken, URL query parameters are always going to be strings. Further, `None` is the default value for the `default` parameter. So the equivalent code is
```suggestion
    first_name = request.args.get("first_name")
    last_name = request.args.get("last_name")
    dod_id = request.args.get("dod_id")
```

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]
Nice! Validating these inputs is going to be really great for the local dev environment. I have a changes to request though. Nice comments to explain the validation process.

**Review by kylelaker-ccpo — APPROVED** [review]
Looks good to me. Since we did some pair programming on parts of this, I would like to get @jasonburkert-ccpo to review as well. Locally validated the same test cases used in the Jira ticket.

**Review by jasonburkert-ccpo — APPROVED** [review]
Good back and forth with @kylelaker-ccpo

**Pull request merged**